public/app/features/commandPalette/CommandPalette.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useEffect, useState } from 'react';
+import {
+  KBarAnimator,
+  KBarPortal,
+  KBarPositioner,
+  KBarResults,
+  KBarSearch,
+  useMatches,
+  Action,
+  VisualState,
+  useRegisterActions,
+  useKBar,
+} from 'kbar';
+import { useStyles2 } from '@grafana/ui';
+import { GrafanaTheme2 } from '@grafana/data';
+import { ResultItem } from './ResultItem';
+import getGlobalActions from './actions/global.static.actions';
+import getDashboardNavActions from './actions/dashboard.nav.actions';
+import { useSelector } from 'react-redux';
+import { StoreState } from 'app/types';
+import { css } from '@emotion/css';
+import { keybindingSrv } from '../../core/services/keybindingSrv';
+
+/**
+ * Wrap all the components from KBar here.
+ * @constructor
+ */
+
+export const CommandPalette = () => {
+  const styles = useStyles2(getSearchStyles);
+  const [actions, setActions] = useState<Action[]>([]);
+  const { showing, query } = useKBar((state) => ({
+    showing: state.visualState !== VisualState.hidden,
+  }));
+
+  const { navBarTree } = useSelector((state: StoreState) => {
+    return {
+      navBarTree: state.navBarTree,
+    };
+  });
+
+  keybindingSrv.bind('esc', () => {
+    if (showing) {
+      query.setVisualState(VisualState.animatingOut);
+    }
+  });
+
+  useEffect(() => {
+    const setGlobalActions = async () => {
+      const staticActions = getGlobalActions(navBarTree);
+      const dashAct = await getDashboardNavActions('go/dashboard');
+      setActions([...staticActions, ...dashAct]);
+    };
+    setGlobalActions();
```

Comment:
nit: 
```suggestion
    (async () => {
      const staticActions = getGlobalActions(navBarTree);
      const dashAct = await getDashboardNavActions('go/dashboard');
      setActions([...staticActions, ...dashAct]);
    })();
```

Reply:
Ah, @ifrost recommended this too but I misunderstood. Fixed!